serde_codegen/src/attr.rs

```diff
@@ -325,3 +411,180 @@ fn get_serde_meta_items(attr: &ast::Attribute) -> Option<&[P<ast::MetaItem>]> {
         _ => None
     }
 }
+
+fn parse_lit_into_expr(cx: &ExtCtxt, name: &str, lit: &ast::Lit) -> Result<P<ast::Expr>, Error> {
+    let s: &str = match lit.node {
+        ast::LitKind::Str(ref s, ast::StrStyle::Cooked) => &s,
+        _ => {
+            cx.span_err(
+                lit.span,
+                &format!("{} literal `{}` must be a string",
+                         name,
+                         lit_to_string(lit)));
+
+            return Err(Error);
+        }
+    };
+
+    let expr = parse::parse_expr_from_source_str("<lit expansion>".to_string(),
```

Comment:
if this fails, the error reporting will be very confusing. is there a way to give a span?